nntrainer/src/conv2d_layer.cpp

```diff
@@ -315,20 +312,32 @@ sharedConstTensors Conv2DLayer::backwarding(sharedConstTensors derivatives,
    *       x [(input_dim.height() + padding[0]*2)
    *           *(input_dim.width() + padding[1]*2) (width)]
    */
+  using uint = unsigned int;
+
+  Tensor ret(in_dim.batch(), in_dim.channel(), in_dim.height() + padding[0] * 2,
+             in_dim.width() + padding[1] * 2);
+  ret.setZero();
 
   TensorDim kdim(ret.channel(), filter_size, kernel_size[0], kernel_size[1]);
 
-  std::vector<float> imkernel(kdim.getDataLen());
+  uint kernel_total_size = kernel_size[0] * kernel_size[1];
 
-  unsigned int count = 0;
-  float *d = imkernel.data();
+  Tensor imKernel(1, 1, in_dim.channel(), filter_size * kernel_total_size);
+  float *d = imKernel.getData();
```

Comment:
rename to imKernel_data or something more verbose?

Reply:
changed to `imKernel_raw`